proposed/http-message.md

```diff
@@ -232,15 +236,177 @@ interface MessageInterface
      * @param string $header Case-insensitive header name.
      * @return string[]
      */
-    public function getHeaderAsArray($header);
+    public function getHeaderLines($header);
+
+    /**
+     * Sets a header, replacing any existing values of any headers with the
+     * same case-insensitive name.
+     *
+     * The header name is case-insensitive. The header values MUST be a string
+     * or an array of strings.
+     *
+     * @param string $header Header name
+     * @param string|string[] $value Header value(s).
+     * @return void
+     * @throws \InvalidArgumentException for invalid header names or values.
+     */
+    public function setHeader($header, $value);
+
+    /**
+     * Appends a header value for the specified header.
+     *
+     * Existing values for the specified header will be maintained. The new
+     * value(s) will be appended to the existing list.
+     *
+     * @param string $header Header name to add
+     * @param string|string[] $value Header value(s).
+     * @return void
+     * @throws \InvalidArgumentException for invalid header names or values.
+     */
+    public function addHeader($header, $value);
+
+    /**
+     * Remove a specific header by case-insensitive name.
+     *
+     * @param string $header HTTP header to remove
+     * @return void
+     */
+    public function removeHeader($header);
+
+    /**
+     * Gets the body of the message.
+     *
+     * @return StreamableInterface|null Returns the body, or null if not set.
+     */
+    public function getBody();
+
+    /**
+     * Sets the body of the message.
+     *
+     * The body MUST be a StreamableInterface object.
+     *
+     * @param StreamableInterface $body Body.
+     * @return void
+     * @throws \InvalidArgumentException When the body is not valid.
+     */
+    public function setBody(StreamableInterface $body);
```

Comment:
I'm not sure we should allow `null`, actually, as it makes the return variable; I'm going to remove that as a valid return value from the getter.